articles/getting-started-with-nodemcu/index.md

```diff
@@ -0,0 +1,140 @@
+
+### Table of contents
+- [Microcontroller](#microcontroller)
+- [NodeMCU](#nodemcu)
+- [Step by Step Guide](#step-by-step-guide)
+- [Conclusion](#conclusion)
+- [Further Reading](#further-reading)
+
+### Microcontroller
+> According to [Wikipedia](https://en.wikipedia.org/wiki/Microcontroller), A microcontroller (MCU for microcontroller unit) is a small computer on a single metal-oxide-semiconductor (MOS) integrated circuit (IC) chip. A microcontroller contains one or more CPUs (processor cores) along with memory and programmable input/output peripherals.
+
+In simple words, a Microcontroller can be called a mini-computer with an in-built CPU processor, Memory, and other I/O peripherals.
+
+Microcontrollers come with different configurations and architectures depending on the application it is used for. It is being used extensively in various industries like Manufacturing, Robotics, Automation, the Internet of Things, and so on.
+
+For example, Microcontrollers are used as embedded devices in appliances like ACs and Washing machines which enables humans to interact with the devices. They contain in-built programs for simplifying human tasks.
+
+### NodeMCU
+In this article, we will be focusing on the NodeMCU microcontroller. NodeMCU stands for Node MicroController Unit. The below image shows a NodeMCU microcontroller board.
+
+![NodeMCU ESP2866](/engineering-education/getting-started-with-nodemcu/nodemcu.png)
+
+[Image Source](https://cityos-air.readme.io/docs/esp8266-nodemcu)
+
+> According to [this](https://components101.com/development-boards/nodemcu-esp8266-pinout-features-and-datasheet) article, NodeMCU is an open-source Lua based firmware and development board specially targeted for IoT based applications. It includes firmware that runs on the ESP8266 Wi-Fi SoC from Espressif Systems, and hardware that is based on the ESP-12 module.
+
+#### Features
+- NodeMCU ESP12 module contains Tensilica Xtensa® 32-bit LX106 RISC microprocessor.
+- Operates at 80 to 160 MHz.
+- Operating voltage of 3.3V.
+- 128KB RAM and 4MB of Flash memory.
+- Contains inbuilt 802.11b/g/n WiFi transceiver.
+- Contains 17 GPIOs (General Purpose Input Output).
+- Provides serial communication via the UART interface.
+
+All the above components with several additional features can be seen in this pin diagram.
+
+![ESP2866 Pin Diagram](/engineering-education/getting-started-with-nodemcu/pin-diagram.png)
+
+[Image source](https://randomnerdtutorials.com/esp8266-pinout-reference-gpios/)
+
+When compared with other microcontrollers like Arduino, we see that:
+- NodeMCU comes with 128KB RAM, whereas Arduino has 2KB RAM.
+- NodeMCU has 4MB ROM, whereas Arduino has 32KB ROM.
+- NodeMCU comes with a MicroUSB port, but Arduino requires a USB-B type.
+- The development board for NodeMCU is much smaller when compared with others.
+- NodeMCU is very cheap with a cost of 5 USD.
```

Comment:
This comparison is pretty great, but it would add more value if you discuss the implications of those. For example,

- NodeMCU comes with a MicroUSB port, but Arduino requires a USB-B type. More devices use MicroUSB and this would ensure better connectivity.